ProcessLocksScreen.c

```diff
@@ -64,18 +65,18 @@ static void ProcessLocksScreen_scan(InfoScreen* this) {
 
          char entry[512];
          if (ULLONG_MAX == data->end) {
-            xSnprintf(entry, sizeof(entry), "%10d  %-10s %-10s %-10s %02x:%02x:%020"PRIu64" %20"PRIu64" %20s  %s",
-               data->id,
+            xSnprintf(entry, sizeof(entry), "%5d %-10s %-10s %-10s %#6"PRIx64" %10"PRIu64" %19"PRIu64" %19s  %s",
+               data->fd,
                data->locktype, data->exclusive, data->readwrite,
-               data->dev[0], data->dev[1], data->inode,
+               (uint64_t) data->dev, data->inode,
                data->start, "<END OF FILE>",
                data->filename ? data->filename : "<N/A>"
             );
          } else {
-            xSnprintf(entry, sizeof(entry), "%10d  %-10s %-10s %-10s %02x:%02x:%020"PRIu64" %20"PRIu64" %20"PRIu64"  %s",
-               data->id,
+            xSnprintf(entry, sizeof(entry), "%5d %-10s %-10s %-10s %#6"PRIx64" %10"PRIu64" %19"PRIu64" %19"PRIu64"  %s",
+               data->fd,
                data->locktype, data->exclusive, data->readwrite,
-               data->dev[0], data->dev[1], data->inode,
+               (uint64_t) data->dev, data->inode,
```

Comment:
Why are you changing the width of the FD column? The width of 10 is intentional, due to the maximum value of `2^31-1` you could have there.

Also I don't think converting the `DEV` field in this way is portable.

Reply:
it is, dev_t is defined to be an integer type, and that's what we already write for lsof output (well, lsof `makedev()`s for us).

likewise, the FD column is 5-wide, like in lsof output. the limit here is not 2^31, it's "whatever the soft ulimit is", which for most systems is a sensible 1024. systemd sets its own ulimit to 1048576, besides that:
```
$ grep 'Max open files' /proc/*/limits | awk '{print $4}' | sort -n | uniq -c
      1 50
      1 256
    684 1024
      1 8192
      1 10000
      2 16384
      3 16424
      8 65535
      9 524288
      6 1048576
```
*If* you saw a ≥10^6 *ever* you'd have gotten reports to widen the FD column in the file listing. But you don't, so you haven't.

Reply:
Also, notably, I didn't change the width of the FD column, I took care to *not* do it, by replacing the 10-wide ID column (which, sure, you could have that many locks, I guess), with the 5-wide FD column copied from the file listing.